### DISCUSSION TITLE
Mount EFS on EKS Bottlerocket node

### DISCUSSION THREAD

**Discussion by nikosarg** [bottlerocket-os/bottlerocket#4094] [Q&A] [answered]
I'm having an EKS cluster where I create node groups using EC2 launch templates.

If I use an Amazon Linux 2(GPU enabled) AMI type, I can specify an EFS mount on the User data of the launch template so each new node will mount the same EFS.
[Amazon EC2 user data](https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html#launch-template-custom-ami
) in launch templates that are used with managed node groups must be in the [MIME multi-part archive] format for Amazon Linux AMIs and TOML format for Bottlerocket AMIs.
I've used the following [user data on EC2 instance launch](https://github.com/bottlerocket-os/bottlerocket/blob/develop/SECURITY_GUIDANCE.md#amazon-ec2) in order the test the TOML format.

I've also read about the [Mount propagations in bootstrap and superpowered containers](https://github.com/bottlerocket-os/bottlerocket?tab=readme-ov-file#mount-propagations-in-bootstrap-and-superpowered-containers).

So, If I use a Bottlerocket NVIDIA AMI type on my node group, I need to do the same, specify an EFS mount in TOML format on the EC2 launch template. Is this even possible?

I've read the following discussions but couldn't find a clear answer:
https://github.com/bottlerocket-os/bottlerocket/discussions/2136
https://github.com/bottlerocket-os/bottlerocket/discussions/1991

Is there another way that I can mount EFS on the nodes?

**Comment by arnaldo2792** [comment, marked as answer]
Hey @nikosarg , thanks for opening this discussion. Out of curiosity, do you create bind mounts in the containers from the path in the host where the EFS filesystem is mounted? Have you tried to use the EFS CSI driver to manage your EFS filesystems?

https://github.com/kubernetes-sigs/aws-efs-csi-driver

**Reply by nikosarg** [reply]
Hi @arnaldo2792, I thought I had to first mount the EFS on the node using the launch template and then having installed the EFS CSI driver on the cluster, to mount the EFS on the Pods with StorageClasses, PersistentVolumeClaims, etc.
That's what I was doing with the Amazon Linux 2(GPU enabled) AMI type nodes but it seems the mount in the launch template user data is totally useless.

I've tested the Bottlerocket NVIDIA AMI type node, having already installed the EFS CSI driver and without specifying anything in the user data of the launch template and it worked fine.

It seems that I was overthinking it a bit. You opened my eyes! Thank you!
